### PR TITLE
twitter handle indexing

### PR DESCRIPTION
**Pull request by imabdulbasit**

### CODE REVIEW

File: crates/core/migrations/2022-03-24-181306_twitter_handle_name_service_table/up.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+create table twitter_handle_name_services (
+    address                      varchar(48)    primary key,
+    wallet_address               varchar(48)    not null,
+    twitter_handle               text           not null                      
+);
```
Comment (kespinola) [inline crates/core/migrations/2022-03-24-181306_twitter_handle_name_service_table/up.sql:5]: We'll want index on wallet_address and twitter_handle.

**Review by kespinola — COMMENTED** [review]
so good. clippy wants some stuff.

File: crates/core/migrations/2022-03-24-181306_twitter_handle_name_service_table/up.sql (hunk 1 above)
Comment (imabdulbasit) [reply crates/core/migrations/2022-03-24-181306_twitter_handle_name_service_table/up.sql:5]: yeah I gotta test this first, I'll add indexes and slot thing to it

File: crates/indexer/src/accountsdb/accounts/name_service.rs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    let values = TwitterHandle {
+        address: Owned(pubkey.clone()),
+        wallet_address: Owned(wallet.to_string()),
+        twitter_handle: Owned(th.handle),
+        slot: slot.try_into()?,
+    };
+
+    match rows.get(0) {
+        Some(indexed) => {
+            if incoming_slot > indexed.slot {
```
Comment (ray-kast) [inline crates/indexer/src/accountsdb/accounts/name_service.rs:50]:
```suggestion
        Some(indexed) if incoming_slot > indexed.slot => {
```

File: crates/indexer/src/accountsdb/accounts/name_service.rs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                            twitter_handle_name_services::table
+                                .filter(twitter_handle_name_services::address.eq(pubkey)),
+                        )
+                        .set(&values)
+                        .execute(db)
+                    })
+                    .await
+                    .context("failed to update twitter handle")?;
+            }
+        },
```
Comment (ray-kast) [inline crates/indexer/src/accountsdb/accounts/name_service.rs:64]:
```suggestion
        },
        Some(_) => (),
```

**Review by ray-kast — CHANGES_REQUESTED** [review]
@imabdulbasit let me know when you get the deserialization error worked out

**Comment by imabdulbasit** [comment]
> @imabdulbasit let me know when you get the deserialization error worked out

Its because of incorrect hashed data so nothing much can be done

**Review by ray-kast — APPROVED** [review]

**Pull request merged**